dandi/support/bids/validator.py

```diff
@@ -50,23 +54,19 @@ def _get_paths(bids_paths):
         ".bidsignore",
         "dandiset.yaml",
     ]
-    # Inelegant hard-coded solution.
-    # Could be replaced by a maximum depth limit if BIDS root auto-detection is implemented.
-    treat_as_file_suffix = [".ngff"]
 
     path_list = []
     for bids_path in bids_paths:
         bids_path = os.path.abspath(os.path.expanduser(bids_path))
         if os.path.isfile(bids_path):
-            path_list.append(bids_path)
+            path_list.append(Path(bids_path).as_posix())
             continue
-        for root, dirs, file_names in os.walk(bids_path, topdown=False):
-            if any(root.endswith(i) for i in treat_as_file_suffix):
-                continue
-            if any(f"{i}/" in root for i in treat_as_file_suffix):
-                continue
-            if any(f"{i}\\" in root for i in treat_as_file_suffix):
-                continue
+        for root, dirs, file_names in os.walk(bids_path, topdown=True):
+            if root.endswith(tuple(pseudofile_suffixes)):
+                # Add the directory name to the validation paths list.
+                path_list.append(Path(root).as_posix() + "/")
+                # Do not index the contents of the directory.
+                dirs[:] = []
```

Comment:
Shouldn't you add a `continue` here so that no file paths in the directory get appended to the path list?

Reply:
it's not needed, when topdown is True, the directory list can be modified in-place (fourth paragraph here: https://docs.python.org/3/library/os.html#os.walk )

Reply:
I meant that the code falls through at line 69 to line 73, so any files at the first level of a directory with a pseudofile suffix will still be added to `path_list`, won't they?

Reply:
ah, yes, you are right. Thanks for picking up on this.